src/rgw/rgw_rados.cc

```diff
@@ -1302,6 +1302,10 @@ int RGWPeriod::update()
     return ret;
   }
 
+  // clear zone short ids of removed zones. period_map.update() will add the
+  // remaining zones back
+  period_map.short_zone_ids.clear();
```

Comment:
@cbodley the problem here I think is that zones that were already assigned an id after collision will have new id assigned to them. I think we need to make a different change in RGWPeriodMap::update(), where we remove entries off short_zone_id if not found.